webapp/src/containers/WalletPage/components/SendPage/index.tsx

```diff
@@ -292,21 +285,23 @@ class SendPage extends Component<SendPageProps, SendPageState> {
       regularDFI,
     });
     if (isAmountValid && isAddressValid) {
-      let amount;
+      let amount: BigNumber;
       let txHash;
       if (
         (!this.tokenSymbol || this.tokenSymbol === 'DFI') &&
-        regularDFI !== 0
+        !regularDFI.isZero()
       ) {
         // Convert to base unit
-        amount = getAmountInSelectedUnit(
+        amount = convertAmountFromUnit(
           this.state.amountToSendDisplayed,
           DEFAULT_UNIT,
           this.props.unit
         );
+        // amount.is
+        const feeCheck = amount.eq(this.props.sendData.walletBalance);
         // if amount to send is equal to wallet balance then cut tx fee from amountToSend
         try {
-          if (new BigNumber(amount).eq(this.props.sendData.walletBalance)) {
+          if (feeCheck) {
             txHash = await sendToAddress(this.state.toAddress, amount, true);
```

Comment:
These 5 lines can be simplified to:

`txHash = await sendToAddress(this.state.toAddress, amount, feeCheck);`

Reply:
Noted, Great find.